src/cdk/collections/array-data-source.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Observable, of as observableOf} from 'rxjs';
+import {DataSource} from './data-source';
+
+
+/** DataSource wrapper for a native array. */
+export class ArrayDataSource<T> extends DataSource<T> {
```

Comment:
Note to reviewers: The `ArrayDataSource` is currently in `cdk`, not `cdk-experimental`. Do we want this or should we move it to experimental?